make/jdk/src/classes/build/tools/generatezip/GenerateZip.java

```diff
@@ -0,0 +1,284 @@
+                    Path filepath = Paths.get(file);
+                    processFiles(filepath);
+                }
+
+                try (FileOutputStream out = new FileOutputStream(fname)) {
+                    boolean createOk = create(new BufferedOutputStream(out, 4096));
+                    if (ok) {
+                        ok = createOk;
+                    }
+                }
+            } else {
+                error("Target zip file "+fname+" already exists.");
+                ok = false;
+            }
+        } catch (IOException e) {
+            fatalError(e);
+            ok = false;
+        } catch (Error ee) {
+            ee.printStackTrace();
+            ok = false;
+        } catch (Throwable t) {
+            t.printStackTrace();
+            ok = false;
+        }
+        return ok;
+    }
+
+    boolean parseArgs(String args[]) {
+        try {
+            boolean parsingIncludes = false;
+            boolean parsingExcludes = false;
+            int count = 0;
+            while(count < args.length) {
+                if (args[count].startsWith("-")) {
+                    String flag = args[count].substring(1);
+                    switch (flag.charAt(0)) {
+                    case 'f':
+                        fname = args[++count];
+                        break;
+                    case 't':
+                        // SOURCE_DATE_EPOCH timestamp specified
+                        timestamp = Long.parseLong(args[++count]) * 1000;
+                        break;
+                    case 'v':
+                        verbose = true;
+                        break;
+                    default:
+                        error(String.format("Illegal option -%s", String.valueOf(flag.charAt(0))));
+                        usageError();
+                        return false;
+                    }
+                } else {
+                    // file or dir to zip
+                    files.add(args[count]);
+                }
+                count++;
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            usageError();
+            return false;
+        } catch (NumberFormatException e) {
+            usageError();
+            return false;
+        }
+        if (fname == null) {
+            error(String.format("-f <archiveName> must be specified"));
+            usageError();
+            return false;
+        }
+        // If no files specified then default to current directory
+        if (files.size() == 0) {
+            error("No input directory or files were specified");
+            usageError();
+            return false;
+        }
+
+        return true;
+    }
+
+    // Walk tree matching files and adding to entries list
+    void processFiles(Path path) throws IOException {
+        File fpath = path.toFile();
+        boolean pathIsDir = fpath.isDirectory();
+
+        // Keep a sorted Set of files to be processed, so that the Jmod is reproducible
```

Comment:
Aren't we generating zip files rather than jmod files here?

Reply:
copy&paste error...